server/leafnode.go

```diff
@@ -177,10 +177,10 @@ func newLeafNodeCfg(remote *RemoteLeafOpts) *leafNodeCfg {
 	if len(remote.DenyExports) > 0 || len(remote.DenyImports) > 0 {
 		perms := &Permissions{}
 		if len(remote.DenyExports) > 0 {
-			perms.Subscribe = &SubjectPermission{Deny: remote.DenyExports}
+			perms.Publish = &SubjectPermission{Deny: remote.DenyExports}
```

Comment:
Originally the DenyExport was a subscribe permission because it meant that if on this LN connection, if we deny export of "foo" it means that it would reject a subscription (hence subscribe permission) on "foo" from the other side. Now you are changing to simply not allowing this server to publish on "foo". I am not saying this is wrong, but we completely flip the meaning.

Reply:
The previous way was wrong. I do honor local deny clauses though now as well.